test/suites/integration/target_group_policy_test.go

```diff
@@ -0,0 +1,130 @@
+		policy = testFramework.CreateTargetGroupPolicy(service, &test.TargetGroupPolicyConfig{
+			PolicyName: "test-policy",
+			Protocol:   aws.String(vpclattice.TargetGroupProtocolHttp),
+		})
+
+		httpRoute = testFramework.NewHttpRoute(testGateway, service, service.Kind)
+
+		testFramework.ExpectCreated(ctx, deployment, service, policy, httpRoute)
+
+		tg := testFramework.GetTargetGroup(ctx, service)
+
+		Expect(*tg.Protocol).To(Equal(vpclattice.TargetGroupProtocolHttp))
+
+		testFramework.UpdateTargetGroupPolicy(policy, &test.TargetGroupPolicyConfig{
+			Protocol: aws.String(vpclattice.TargetGroupProtocolHttps),
+		})
+
+		testFramework.ExpectUpdated(ctx, policy)
+
+		httpsTG := testFramework.GetTargetGroupWithProtocol(ctx, service, "https", "http1")
+
+		Expect(*httpsTG.Protocol).To(Equal(vpclattice.TargetGroupProtocolHttps))
+	})
+
+	It("Delete Target Group Policy reset health check config for HTTP and HTTP1 Target Group", func() {
+		policy = testFramework.CreateTargetGroupPolicy(service, &test.TargetGroupPolicyConfig{
+			PolicyName:      "test-policy",
+			Protocol:        aws.String(vpclattice.TargetGroupProtocolHttp),
+			ProtocolVersion: aws.String(vpclattice.TargetGroupProtocolVersionHttp1),
+			HealthCheck: &v1alpha1.HealthCheckConfig{
+				IntervalSeconds: aws.Int64(7),
+				StatusMatch:     aws.String("200,204"),
+			},
+		})
+
+		httpRoute = testFramework.NewHttpRoute(testGateway, service, service.Kind)
+
+		testFramework.ExpectCreated(ctx, deployment, service, policy, httpRoute)
+
+		tgSummary := testFramework.GetTargetGroup(ctx, service)
+
+		tg := testFramework.GetFullTargetGroupFromSummary(ctx, tgSummary)
+
+		Expect(*tg.Config.HealthCheck.ProtocolVersion).To(Equal(vpclattice.TargetGroupProtocolVersionHttp1))
+		Expect(*tg.Config.HealthCheck.Protocol).To(Equal(vpclattice.TargetGroupProtocolHttp))
+		Expect(*tg.Config.HealthCheck.HealthCheckIntervalSeconds).To(BeEquivalentTo(7))
+		Expect(*tg.Config.HealthCheck.Matcher.HttpCode).To(Equal("200,204"))
+
+		testFramework.ExpectDeletedThenNotFound(ctx, policy)
+
+		time.Sleep(2 * time.Second)
+
+		updatedTG := testFramework.GetFullTargetGroupFromSummary(ctx, tgSummary)
+
+		Expect(*updatedTG.Config.HealthCheck.Enabled).To(BeTrue())
+		Expect(*updatedTG.Config.HealthCheck.Path).To(Equal("/"))
+		Expect(*updatedTG.Config.HealthCheck.HealthCheckIntervalSeconds).To(BeEquivalentTo(30))
+		Expect(*updatedTG.Config.HealthCheck.HealthCheckTimeoutSeconds).To(BeEquivalentTo(5))
+		Expect(*updatedTG.Config.HealthCheck.HealthyThresholdCount).To(BeEquivalentTo(5))
+		Expect(*updatedTG.Config.HealthCheck.UnhealthyThresholdCount).To(BeEquivalentTo(2))
+		Expect(*updatedTG.Config.HealthCheck.Protocol).To(Equal(vpclattice.TargetGroupProtocolHttp))
+		Expect(*updatedTG.Config.HealthCheck.ProtocolVersion).To(Equal(vpclattice.TargetGroupProtocolVersionHttp1))
+		Expect(*updatedTG.Config.HealthCheck.Matcher.HttpCode).To(Equal("200"))
+		Expect(updatedTG.Config.HealthCheck.Port).To(BeNil())
```

Comment:
Can you compare structs rather than each field of struct? Also when we reset health check config there should be some DefaultHealthCheckConfig struct in source code that we should compare with, otherwise we declare default behavior in 2 different places source and test.

Reply:
Default Health Check config is different based on the Target Group Protocol (it's a private method that takes in `TargetGroupProtocol`)

Reply:
ok, you can make it public, also dont compare each field of struct but compare structs

Reply:
Since each field of a struct is a pointer, I can't use `Equal` to compare them.

Also for the default health check config method, it returned a struct with a "reset" value of `0` for each field instead the actual value, it's not possible to use that.

e.g. Our `UpdateTargetGroup` operation takes the following request:

```json
{
  "healthCheck": {
    "healthCheckIntervalSeconds": 0
  }
}
```

and then GetTargetGroup operation on the same target group will return

```json
{
  "healthCheck": {
    "healthCheckIntervalSeconds": 5
  }
}
```

Reply:
> Since each field of a struct is a pointer, I can't use Equal to compare them.

Are you sure? assert.Equals de-reference pointers. Works fine for me

```go
type A struct {
	i int
	s string
	b *B
}

type B struct {
	i int
	s string
}

func TestA(t *testing.T) {
	a1 := &A{
		i: 10,
		s: "10",
		b: &B{
			i: 20,
			s: "20",
		},
	}
	a2 := &A{
		i: 10,
		s: "10",
		b: &B{
			i: 20,
			s: "20",
		},
	}
	assert.Equal(t, a1, a2)
}
m
```

Reply:
reflect.DeepEqual de-reference pointers https://pkg.go.dev/reflect#DeepEqual

> Pointer values are deeply equal if they are equal using Go's == operator or if they point to deeply equal values. 

Code above also resolves into True, with reflect.DeepEqual(a1, a2)